### PR TITLE
TREV-1138 Valmistava opetus käyttöön kehyskunnissa

### PR DESCRIPTION
**Pull request by cyanidesayonara**
Enable preparatory education in employee UI for Hämeenkyrö & Vesilahti

https://voltti.atlassian.net/browse/TREV-1138

Screenshot 2024-02-26 160945

### CODE REVIEW

File: frontend/vesilahti/employee.tsx [hunk 1]
```diff
@@ -607,6 +607,9 @@ const customizations: EmployeeCustomizations = {
     'DAYCARE',
     'TEMPORARY_DAYCARE',
     'PRESCHOOL_DAYCARE_ONLY',
+    'PREPARATORY',
+    'PREPARATORY_DAYCARE',
+    'PREPARATORY_DAYCARE_ONLY',
```
Comment (patari) [inline frontend/vesilahti/employee.tsx:612]: There is only `PRESCHOOL_DAYCARE_ONLY` so i guess this should also be just `PREPARATORY_DAYCARE_ONLY`.

File: service/src/main/resources/WEB-INF/templates/vesilahti/preparatory/decision.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<!--
+SPDX-FileCopyrightText: 2024 Tampere region
+
+SPDX-License-Identifier: LGPL-2.1-or-later
+-->
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org">
+  <head>
+    <meta charset="utf-8"/>
+    <style th:replace="~{hameenkyro/shared/common.html :: css}"></style>
```
Comment (patari) [inline service/src/main/resources/WEB-INF/templates/vesilahti/preparatory/decision.html:10]: `hameenkyro` -> `vesilahti`. Also lines 15, 16, 19, ...

**Review by patari — CHANGES_REQUESTED** [review]

File: frontend/vesilahti/employee.tsx (hunk 1 above)
Comment (cyanidesayonara) [reply frontend/vesilahti/employee.tsx]: PREPARATORY & PREPARATORY_DAYCARE removed

File: service/src/main/resources/WEB-INF/templates/vesilahti/preparatory/decision.html (hunk 2 above)
Comment (cyanidesayonara) [reply service/src/main/resources/WEB-INF/templates/vesilahti/preparatory/decision.html]: Copy/paste error fixed

File: service/src/main/resources/db/data/hameenkyro/afterMigrate__service_need_options.sql [hunk 3]
```diff
@@ -17,6 +17,7 @@ VALUES
...
     ('e1063bee-c19d-469d-85a5-6b0350872d76', 'Tilapäinen varhaiskasvatus', 'Tilapäinen varhaiskasvatus', 'Temporary daycare', 'TEMPORARY_DAYCARE', TRUE, 1.0, 1.0, 1.75, 1.0, 1.75, 40, NULL, NULL, FALSE, FALSE, 'Tilapäinen varhaiskasvatus', 'Tilapäinen varhaiskasvatus', 'Tilapäinen varhaiskasvatus', 'Tilapäinen varhaiskasvatus', TRUE, TRUE, NULL),
     ('94e44ef1-106b-401d-81b6-8e5c31cd0437', 'Esiopetus', 'Esiopetus', 'Preschool', 'PRESCHOOL', TRUE, 0.0, 0.5, 0.5, 0.5, 0.5, 0, NULL, NULL, TRUE, TRUE, 'Esiopetus', 'Esiopetus', 'Esiopetus', 'Esiopetus', TRUE, TRUE, NULL),
+    ('1a4b0997-b83d-44ec-8bd9-98b12e5d6d04', 'Valmistava opetus', 'Valmistava opetus', 'Preparatory education', 'PREPARATORY', TRUE, 0.0, 0.5, 0.5, 0.5, 0.5, 0, NULL, NULL, TRUE, TRUE, 'Valmistava opetus', 'Valmistava opetus', 'Valmistava opetus', 'Valmistava opetus', TRUE, TRUE, NULL),
```
Comment (patari) [inline service/src/main/resources/db/data/hameenkyro/afterMigrate__service_need_options.sql:20]: This is still missing options for `PREPARATORY_DAYCARE` & `PREPARATORY_DAYCARE_ONLY`, good starting point is to copy from `PRESCHOOL_DAYCARE` & `PRESCHOOL_DAYCARE_ONLY`.

File: service/src/main/resources/db/data/vesilahti/afterMigrate__service_need_options.sql [hunk 4]
```diff
@@ -12,6 +12,7 @@ VALUES
...
     ('503590f0-b961-11eb-b520-53740af3f7ef', 'Varhaiskasvatus 121-150 tuntia / kuukausi', 'Varhaiskasvatus 121-150 tuntia / kuukausi', 'Daycare - 121-150 h per month', 'DAYCARE', FALSE, 1.0, 1.0, 1., 1.0, 1.0, 40, NULL, 150, TRUE, TRUE, 'Varhaiskasvatus 121-150 tuntia / kuukausi', 'Varhaiskasvatus 121-150 tuntia / kuukausi', 'Varhaiskasvatus 121-150 tuntia / kuukausi', 'Varhaiskasvatus 121-150 tuntia / kuukausi', TRUE, TRUE, 2),
     ('503591ae-b961-11eb-b521-1fca99358eed', 'Varhaiskasvatus yli 151 tuntia / kuukausi', 'Varhaiskasvatus yli 151 tuntia / kuukausi', 'Daycare - over 151 h per month', 'DAYCARE', FALSE, 1.0, 1.0, 1.0, 1.0, 1.0, 40, NULL, 210, FALSE, FALSE, 'Varhaiskasvatus yli 151 tuntia / kuukausi', 'Varhaiskasvatus yli 151 tuntia / kuukausi', 'Varhaiskasvatus yli 151 tuntia / kuukausi', 'Varhaiskasvatus yli 151 tuntia / kuukausi', TRUE, TRUE, 3),
+    ('b0edf1ff-7b36-43b2-9577-e284d6bbc964', 'Valmistava opetus', 'Valmistava opetus', 'Preparatory education', 'PREPARATORY', TRUE, 0.0, 0.5, 0.5, 0.5, 0.5, 0, NULL, NULL, TRUE, TRUE, 'Valmistava opetus', 'Valmistava opetus', 'Valmistava opetus', 'Valmistava opetus', TRUE, TRUE, NULL),
```
Comment (patari) [inline service/src/main/resources/db/data/vesilahti/afterMigrate__service_need_options.sql:15]: There is no `PREPARATORY`, only `PREPARATORY_DAYCARE_ONLY`...

**Review by patari — CHANGES_REQUESTED** [review]

File: service/src/main/resources/db/data/hameenkyro/afterMigrate__service_need_options.sql (hunk 3 above)
Comment (cyanidesayonara) [reply service/src/main/resources/db/data/hameenkyro/afterMigrate__service_need_options.sql:20]: Sorry, I still get these terms mixed up... 

Added
* PREPARATORY_DAYCARE -  'Valmistava opetus ja liittyvä varhaiskasvatus', 'Preparatory education'
* PREPARATORY_DAYCARE_ONLY - 'Valmistavan opetuksen liittyvä varhaiskasvatus', 'Preparatory education'

(Couldn't find any other English translation besides 'Preparatory education')

File: service/src/main/resources/db/data/vesilahti/afterMigrate__service_need_options.sql (hunk 4 above)
Comment (cyanidesayonara) [reply service/src/main/resources/db/data/vesilahti/afterMigrate__service_need_options.sql]: Changed to PREPARATORY_DAYCARE_ONLY - 'Valmistavan opetuksen liittyvä varhaiskasvatus', 'Preparatory education'

**Review by patari — APPROVED** [review]

**Pull request merged**